src/System.Memory/tests/Span/Clear.cs

```diff
@@ -198,6 +199,31 @@ public static void ClearReferenceTypeLonger()
             Assert.Equal<string>(expected, actual);
         }
 
+        [Fact]
+        public static void ClearReferenceTypeSlice()
+        {
+            // A string array [ ""1", ..., "20" ]
+            string[] baseline = Enumerable.Range(1, 20).Select(i => i.ToString()).ToArray();
+
+            for (int i = 0; i < 16; i++)
+            {
+                // Going to clear array.Slice(1, i) manually,
+                // then compare it against array.Slice(1, i).Clear().
+                // Test is written this way to allow detecting overrunning bounds.
```

Comment:
We have a bunch of tests that check out-of-range changes (look for "ChecksGoOutOfRange" within test names). For example: https://github.com/dotnet/corefx/blob/master/src/System.Memory/tests/ReadOnlySpan/StartsWith.char.cs#L140

Can we follow a similar approach?

Essentially, we only clear a slice of the span, and assert that the canary value right before/after remains un-changed.

Reply:
I saw those; those canary patterns aren't really applicable to the implementation of this method because it jumps around so much while overwriting values. You'd have to check multiple canaries on either side of the subspan, and at that point you essentially end up with the test code as it's already implemented.

Reply:
> those canary patterns aren't really applicable to the implementation of this method because it jumps around so much while overwriting values. 

Fair point.